docs/usage.rst

```diff
@@ -6,19 +6,30 @@ To use Machetli, you have to write two Python scripts:
 * an evaluator script that checks if the behavior you are
   trying to isolate is still present in a state, and
 * a search script that tells Machetli how to explore the
-  space of intance modifications.
+  space of instance modifications.
 
 .. _usage-evaluator:
 
 Writing an evaluator script
 ---------------------------
 
 The evaluator script is run for each state to check if the desired behavior (for
-example, the bug we are trying to find) is still present after some modifications
-of the instance. For technical reasons, it has to be implemented in its own
-Python file. To write an evaluator, simply create a new Python file and
-implement a function called :meth:`evaluate` in it that takes a single argument (the
-state to evaluate) and returns ``True`` if the behavior is present. Here is a
+example, the bug we are trying to find) is still present after some
+modifications of the instance. For technical reasons, it has to be implemented
+in its own Python file. An evaluator is any script that takes the path to a
+pickled state as its command line argument, and exits with
+:attr:`EXIT_CODE_IMPROVING <machetli.evaluator.EXIT_CODE_IMPROVING>` if the
```

Comment:
The reason we call them "improving/not improving" internally is also not that clear. Guess it's because the search is a hill-climbing search, but it wouldn't have to be. So I'm fine with changing the names but this might be a larger operation because we use those terms a lot. I think 'success' and 'failure' are better than the current terms but I'm not completely happy with them because 'failure' could also mean a crash of the evaluator or running out of resources whereas 'success' could also mean 'successfully completing the evaluator' (independent of the result). Maybe some shorter form of 'behavior_present' and 'behavior_not_present'? or something involving 'detected', 'recognized', ...?

Reply:
We can also do this in a separate issue to not clutter this change too much, but if there's no good reason for "improving/not improving", then I would like to change it. I agree that the suggested names are not good, though. I don't mind 'behavior_[not_]present' and like it more than something involving detected or recognized.

Reply:
We opened issue #97 to do this.